### PR TITLE
Issue 47 notification on failure

### PR DESCRIPTION
**Pull request by juanmatias**
## what
* Retry on merge failure
  * when `git pull` from upstream fails `git pull --rebase` is run instead
  * if this success then the `git push` with the adding of flag `-f` is run
* Notification to Slack
  * When the process fails a Slack notif is sent
  * Also is possible to send notifications on success

## why
* Due to the reasons stated in the corresponding issue https://github.com/binbashar/le-dev-tools/issues/47

## references
* closes https://github.com/binbashar/le-dev-tools/issues/47

### CODE REVIEW

**Review by exequielrafaela — APPROVED** [review]

**Comment by juanmatias** [comment]
Reports:

* `binbashar/terraform-aws-config` switched master to main 
* `binbashar/terraform-aws-cloudfront-s3-cdn` switched master to main 
* `binbashar/terraform-terraform-label` switched master to main

**Comment by juanmatias** [comment]
* `binbashar/terraform-null-label/branches` switched master to main
* `binbashar/terraform-aws-vpc-base` switched master to main

Regarding `terraform-aws-transit-gateway` is being synced from two different repos:
```
"terraform-aws-transit-gateway","cloudposse/terraform-aws-transit-gateway","master" \
"terraform-aws-transit-gateway","terraform-aws-modules/terraform-aws-transit-gateway","master" \
```
Due to this order in the action file, the second one won´t succeed (when the script tries to add an `upstream` remote it fails the second time since it already exists). So I will comment out the second line and change the default branch on our fork to main.

**Comment by juanmatias** [comment]
* `binbashar/terraform-aws-sso` switched master to main
* `binbashar/terraform-aws-secure-baseline` switched master to main
* `binbashar/terraform-aws-sns-topic` switched master to main
* `binbashar/terraform-aws-s3-log-storage` switched master to main

**Comment by juanmatias** [comment]
* `binbashar/terraform-aws-route53-query-logs` switched master to main
* `binbashar/terraform-aws-route53-alias` switched master to main
* `binbashar/terraform-aws-root-login-notifications` switched master to main

**Comment by juanmatias** [comment]
* `binbashar/terraform-aws-multi-az-subnets` switched master to main
* `binbashar/terraform-aws-msk-apache-kafka-cluster` switched master to main
* `binbashar/terraform-aws-logs` switched master to main
* `binbashar/terraform-aws-kms-key` switched master to main
* `binbashar/terraform-aws-firewall-manager` switched master to main

**Comment by juanmatias** [comment]
* `binbashar/terraform-aws-elasticache-redis` switched master to main
* `binbashar/terraform-aws-dynamodb` switched master to main
* `binbashar/terraform-aws-cloudtrail-s3-bucket` switched master to main
* `binbashar/terraform-aws-cloudtrail-cloudwatch-alarms` switched master to main
* `binbashar/terraform-aws-cloudtrail/branches` switched master to main

**Comment by juanmatias** [comment]
* `binbashar/terraform-aws-ses` switched master to main

**Review by diego-ojeda-binbash — APPROVED** [review]

File: .circleci/config.yml [hunk 1]
```diff
@@ -269,7 +274,28 @@ workflows:
...
+  #
+  manual_nightly_sync_and_release:
+    when:  << pipeline.parameters.manual_trigger_parameter >>
+    jobs:
+      - git-sync-fork-upstream:
+          context: binbashar-org-cron-context
+          filters:
+            branches:
+             only: # only branches matching the below regex filters will run
+               - ISSUE-47-notification-on-failure
```
Comment (diego-ojeda-binbash) [inline .circleci/config.yml:293]: ⚠️

File: Makefile [hunk 2]
```diff
@@ -16,6 +18,7 @@ init-makefiles: ## initialize makefiles
 	mkdir -p ${MAKEFILES_DIR}
 	git clone https://github.com/binbashar/le-dev-makefiles.git ${MAKEFILES_DIR} -q
 	cd ${MAKEFILES_DIR} && git checkout ${MAKEFILES_VER} -q
+	git checkout ISSUE-47-le-dev-tools-notification-on-failure
```
Comment (diego-ojeda-binbash) [inline Makefile:21]: ⚠️

Comment (juanmatias) [reply Makefile:21]: this is only for test, removed before the merge

**Pull request merged**